Objects/unionobject.c

```diff
@@ -330,7 +334,22 @@ union_parameters(PyObject *self, void *Py_UNUSED(unused))
     return Py_NewRef(alias->parameters);
 }
 
+static PyObject *
+union_name(PyObject *Py_UNUSED(self), void *Py_UNUSED(ignored))
+{
+    return PyUnicode_FromString("Union");
+}
+
+static PyObject *
+union_origin(PyObject *Py_UNUSED(self), void *Py_UNUSED(ignored))
+{
+    return Py_NewRef(&_PyUnion_Type);
+}
+
 static PyGetSetDef union_properties[] = {
+    {"__name__", union_name, NULL, "Name of the type", NULL},
+    {"__qualname__", union_name, NULL, "Qualified name of the type", NULL},
```

Comment:
For something that does not live in builtins, should `__qualname__` return `typing.Union` instead of just `Union`?

Reply:
No, `__qualname__` is the name plus the name of enclosing classes. For objects defined at the top level in a module, it's the same as the `__name__`.